clients/githubrepo/workflows.go

```diff
@@ -31,15 +35,85 @@ type workflowsHandler struct {
 	repo   string
 }
 
+var (
+	errWorkflowEmptyID   = errors.New("workflow ID is empty")
+	errWorkflowEmptyName = errors.New("workflow Name is empty")
+	errWorkflowEmptyPath = errors.New("workflow Path is empty")
+)
+
 func (handler *workflowsHandler) init(ctx context.Context, owner, repo string) {
 	handler.ctx = ctx
 	handler.owner = owner
 	handler.repo = repo
 }
 
-func (handler *workflowsHandler) listSuccessfulWorkflowRuns(filename string) ([]clients.WorkflowRun, error) {
+func (handler *workflowsHandler) GetWorkflowByFileName(filename string) (clients.Workflow, error) {
+	workflow, _, err := handler.client.Actions.GetWorkflowByFileName(
+		handler.ctx, handler.owner, handler.repo, filename)
+	if err != nil {
+		return clients.Workflow{}, sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("GetWorkflowByFileName: %v", err))
+	}
+	if workflow.ID == nil {
+		return clients.Workflow{}, sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("GetWorkflowByFileName: %v", errWorkflowEmptyID))
+	}
+	if workflow.Name == nil {
+		return clients.Workflow{}, sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("GetWorkflowByFileName: %v", errWorkflowEmptyName))
+	}
+	if workflow.Path == nil {
+		return clients.Workflow{}, sce.WithMessage(sce.ErrScorecardInternal, fmt.Sprintf("GetWorkflowByFileName: %v", errWorkflowEmptyPath))
+	}
+
+	return clients.Workflow{
+		ID:   *workflow.ID,
+		Path: *workflow.Path,
+		Name: *workflow.Name,
+	}, nil
+}
+
+func addOptions(s string, opts interface{}) (string, error) {
+	v := reflect.ValueOf(opts)
+	if v.Kind() == reflect.Ptr && v.IsNil() {
+		return s, nil
+	}
+
+	u, err := url.Parse(s)
+	if err != nil {
+		return s, err
+	}
+
+	qs, err := query.Values(opts)
+	if err != nil {
+		return s, err
+	}
+
+	u.RawQuery = qs.Encode()
+	return u.String(), nil
+}
+
+func (handler *workflowsHandler) listWorkflowRuns(opts *clients.ListWorkflowRunOptions) ([]clients.WorkflowRun, error) {
+	endpoint := fmt.Sprintf("repos/%s/%s/actions/runs", handler.owner, handler.repo)
+	u, err := addOptions(endpoint, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := handler.client.NewRequest("GET", u, nil)
```

Comment:
go-github does not support `check_suite_id` as input https://github.com/google/go-github/blob/178169fc04bc4a05ca48f98709cdd02ab5b556e3/github/actions_workflow_runs.go#L56

So I've simply copied their code from https://github.com/google/go-github/blob/178169fc04bc4a05ca48f98709cdd02ab5b556e3/github/actions_workflow_runs.go#L91 until they add support. We should file an issue for it.

Reply:
Could you file an issue and add a TODO comment here to point to it?